lib/ansible/modules/web_infrastructure/wildfly.py

```diff
@@ -0,0 +1,309 @@
+        # case
+
+        pass
+import ansible.module_utils.six.moves.urllib.error as urllib_error
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url, url_argument_spec
+from ansible.module_utils.pycompat24 import get_exception
+
+
+def undeploy(deployment, server_groups):
+    deployment = '{}'.format(deployment)
+    remove_content = {'operation': 'remove',
+                      'address': [{'deployment': deployment}]}
+
+    if mode == 'domain':
+        server_groups = '{}'.format(server_groups)
+        remove_from_server = {'operation': 'remove',
+                              'address': [{'server-group': server_groups},
+                                          {'deployment': deployment}],
+                              'json.pretty': 1}
+    elif mode == 'standalone':
+        remove_from_server = {'operation': 'remove',
+                              'address': [{'deployment': deployment}],
+                              'json.pretty': 1}
+
+    response = get_response(remove_from_server)
+    if response.code is 200:
+        return get_response(remove_content)
+    else:
+        return response
+
+
+def deploy(deployment, server_groups, src):
+    deployment = '{}'.format(deployment)
+    server_groups = '{}'.format(server_groups)
+    add_content = {'operation': 'add',
+                   'address': [{'deployment': deployment}],
+                   'content': [{'url': 'file:' + src + ''}],
+                   'json.pretty': 1}
+
+    response_add_content = get_response(add_content)
+
+    if response_add_content.code is 200:
+        if mode == 'domain':
+            add_content_server_group = \
+                {'operation': 'add',
+                 'address': [{'server-group': server_groups},
+                             {'deployment': deployment}],
+                 'content': [{'url': 'file:' + src + ''}],
+                 'json.pretty': 1}
+            deploy_content_domain = \
+                {'operation': 'deploy',
+                 'address': [{'server-group': server_groups},
+                             {'deployment': deployment}],
+                 'json.pretty': 1}
+            response_add_content_server_group = \
+                get_response(add_content_server_group)
+            if response_add_content_server_group.code is 200:
+                return get_response(deploy_content_domain)
+            else:
+                return response_add_content_server_group
+        elif mode == 'standalone':
+            deploy_content_standalone = \
+                {'operation': 'deploy',
+                 'address': [{'deployment': deployment}],
+                 'json.pretty': 1}
+            return get_response(deploy_content_standalone)
+    else:
+        return response_add_content
+
+
+def get_response(request_data):
+    headers = {'Content-Type': 'application/json'}
+    response = ''
+    try:
+        response = open_url(
```

Comment:
You should use [`fetch_url`](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/urls.py#L993) instead which encapsulates the `open_url`. That gives you the ability to use other standard parameters like `url_username`, `url_password`, `force_basic_auth`, ... without the need to define it in your code.